python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -1197,6 +1221,115 @@ async def set_hexapod(self, azimuth, elevation, axis=f'xyzuvw'):
                 # correction completed... flip bit on detailedState
                 self.detailed_state = self.detailed_state ^ status_bit
 
+    async def set_atspectrograph_corrections(self, azimuth, elevation):
+        """Utility to apply hexapod and/or pointing corrections based on ATSpectrograph configuration.
+
+        Warning
+        -------
+
+        Can only be applied if hexapod correction is also applied
+
+        Parameters
+        ----------
+        """
+        self.log.debug(f"At the beginning of set_atspectrograph_corrections")
+
+        if self.can_move():
+            # publish new detailed state
+
+            self.log.debug(f"Applying Spectrograph Corrections, if required")
+
+            status_bit = DetailedState.ATSPECTROGRAPH
+            _offsetValue = 0.0
+            _pointingTolerance = 0.01  # arcseconds - arbitrary, just here for rounding issues
+            _pointingOffsets = np.zeros((2))
+
+            # Start with Focus offset
+
+            # check that offset is not just numerical noise. In our case anything smaller than
+            # the correction tolerance won't be applied to the hexapod anyways, so any offset
+            # 10x smaller than that is surely noise.
+
+            if abs(self.focusOffsetYetToBeApplied) > abs(self.correction_tolerance['z'] / 10):
+                self.log.info(
+                    f'Applying focus offset of {self.focusOffsetYetToBeApplied} '
+                    f'in correction loop due to filter '
+                    f'and/or disperser changes.')
+                # add the offset, then reset the value
+                # using subtraction here to avoid a possible race condition
+                # note that self.focusOffsetYetToBeApplied is a value not an object
+                # so no deepcopy is required
```

Comment:
If `deepcopy` is not required why are you using it?

Reply:
fixed..